source/manual/rabbitmq.html.md

```diff
@@ -4,147 +4,123 @@ title: Manage RabbitMQ
+[publishes a message][publishing_api_publishes_message] to our main exchange,
+`publishing_documents`.
+
+**Queues** in the AMQP model queues are very similar to queues in other message
+and task-queueing systems: they store messages that are consumed by
+applications. An example of a queue is `email_alert_service` which is used by
+[email-alert-service][email-alert-service] to forward publishing activity to
+email-alert-api. Queues are [created by consumer
+applications][create_queues].
+
+**Bindings** are rules that exchanges use (among other things) to route
+messages to queues. To instruct an exchange E to route messages to a queue Q, Q
+has to be bound to E. Bindings may have an optional routing key attribute. An
+example of a binding is the `cache_clearing_service-high` queue is bound to the
+`published_documents` exchange with a routing key matching of `*.major`. E.g
+messages sent to the exchange with a routing key of `guide.major` will be
+routed to that queue.
+
+**Messages** consist of a JSON payload and publish options (we predominantly
+use content type, routing key and persistant).
+
+Options:
+
+* content_type (string) - tells the consumer the type of message. E.g
+  `"application/json"`
+* routing_key (string) - matches against bindings to filter messages to certain
+  queues. E.g `"guide.major"`
+* persistant (boolean) - tells RabbitMQ whether to save the message to disk.
+
+Message options are set when a message is published. In our use case, the
+message's payload is the content item in JSON format. The code in the
+publishing-api to publish a message is [here][publish_message_call].
+
+**Consumer applications** - like [email-alert-service][email-alert-service] are
+responsible for configuring queues, examples of this configuration is
+[here][rabbit_config_yml] and [here][rabbit_config_rake]. For
+email-alert-service we consume messages by [running this rake
+task][message_processors] and use the [major change
+processor][major_message_processor] to process the consumed messages. All our
+consumer applications use the [govuk_message_queue_consumer][message_consumer]
+gem to consume messages from RabbitMQ in a standardised way.
+
+You can see live examples of things like queues, exchanges, bindings etc by
+connecting to the RabbitMQ control panel.
+
+### Federated `published_documents` exchange
+
+A federated exchange is connected to an upstream exchanges by AMQP and the
+messages published to the upstream exchanges are copied over to the federated
+exchange.
+
+While the migration of GOV.UK from Carrenza to AWS is ongoing, we run two RabbitMQ clusters,
+one in each environment.
+
+The `published_documents` exchange is federated in both directions, i.e. the RabbitMQ cluster
+from each provider connects as a client to the exchange in the other provider and forwards
+messages to its own exchange.
+
+There is no infinite loop because `max-hops` is set to `1`.
```

Comment:
Minor: all these paragraphs just make the doc look longer. We could combine some of them.